### PR TITLE
Additional Testing for TFWS

### PR DESCRIPTION
**Pull request by tblackwe**
## Description
Fixing the tests for fetch/update/create, as well as adding in unit tests for s3-lib

We are missing some coverage on the Server error testing that I want to clear up, but this PR needs to get moving. 

### How to test
app-lib/yarn test

### Changed Dependencies
None

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [ ] I have added analytics, if necessary
- [ ] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,36 +54,42 @@ In the root of the project run `./dev local`
...
 COGNITO_REDIRECT_SIGNOUT=http://localhost:3000/
-S3_ATTACHMENTS_BUCKET_REGION=us-east-1
+COGNITO_REGION=us-east-1
+COGNITO_USER_POOL_CLIENT_ID=4n2andd7qumjgdojec3cbqsemu
+COGNITO_USER_POOL_CLIENT_DOMAIN=main-login-4n2andd7qumjgdojec3cbqsemu.auth.us-east-1.amazoncognito.com
+COGNITO_USER_POOL_ID=us-east-1_lerDvs4wn
+DEV_API_URL=https://umu6q0vjmk.execute-api.us-east-1.amazonaws.com/main
+LD_PROJECT_KEY=mdct-mcr
+LOCAL_LOGIN=true
+REACT_APP_LD_SDK_CLIENT=placeholder
```
Comment (karla-vm) [inline README.md:90]:
```suggestion
REACT_APP_LD_SDK_CLIENT=63581e3edf884e5943647584
```

File: services/ui-src/src/components/cards/EntityCard/EntityCard.test.tsx [hunk 2]
```diff
@@ -127,6 +155,42 @@ describe("Test AccessMeasures EntityCard accessibility", () => {
...
+    expect(screen.getByTestId("print-status-indicator")).toBeVisible();
+  });
+
+  test("Finished EntityCard in print version displays completed ", () => {
+    expect(screen.getByText("Complete")).toBeVisible();
+  });
+});
+
+describe("Test Uncmpleted Print Version EntityCard", () => {
+  beforeEach(() => {
```
Comment (britt-mo) [inline services/ui-src/src/components/cards/EntityCard/EntityCard.test.tsx:181]: nit: typo

**Review by britt-mo — APPROVED** [review]

File: services/app-api/handlers/reports/fetch.ts [hunk 3]
```diff
@@ -17,31 +17,33 @@ export const fetchReport = handler(async (event, _context) => {
     TableName: process.env.MCPAR_REPORT_TABLE_NAME!,
     Key: { state, id: reportId },
   };
-  const response = await dynamoDb.get(reportMetadataParams);
-  const reportMetadata: any = response.Item; // TODO: strict typing
-  const { formTemplateId, fieldDataId } = reportMetadata;
+  try {
+    const response = await dynamoDb.get(reportMetadataParams);
+    if (!response?.Item) throw "Record not found in database";
```
Comment (karla-vm) [inline services/app-api/handlers/reports/fetch.ts:22]: Might want to move these to the `errors.ts` file and pull them in.

File: services/app-api/handlers/reports/update.ts [hunk 4]
```diff
@@ -39,89 +39,97 @@ export const updateReport = handler(async (event, context) => {
...
-        // get formTemplate from s3 bucket (for passed fieldData validation)
-        const formTemplateParams = {
-          Bucket: process.env.MCPAR_FORM_BUCKET!,
-          Key: "formTemplates/" + state + "/" + formTemplateId,
-        };
-        const formTemplate: any = await s3Lib.get(formTemplateParams); // TODO: strict typing
+          // get existing fieldData from s3 bucket (for patching with passed data)
+          const fieldDataParams = {
+            Bucket: process.env.MCPAR_FORM_BUCKET!,
+            Key: "fieldDatas/" + state + "/" + fieldDataId,
```
Comment (karla-vm) [inline services/app-api/handlers/reports/update.ts:57]:
```suggestion
            Key: `fieldData/${state}/${fieldDataId}`,
```

typo here on `fieldData` (and maybe we change to literals for consistency with `create`

File: services/app-api/handlers/reports/update.ts [hunk 5]
```diff
@@ -39,89 +39,97 @@ export const updateReport = handler(async (event, context) => {
...
+            if (validatedFieldData) {
+              // post validated field data to s3 bucket
+              const fieldData = {
+                ...existingFieldData,
+                ...validatedFieldData,
               };
-              await dynamoDb.put(reportMetadataParams);
+              const fieldDataParams = {
+                Bucket: process.env.MCPAR_FORM_BUCKET!,
+                Key: "fieldData/" + state + "/" + fieldDataId,
```
Comment (karla-vm) [inline services/app-api/handlers/reports/update.ts:83]:
```suggestion
                Key: `fieldData/${state}/${fieldDataId}`,
```

**Review by karla-vm — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (braxex) [reply README.md:90]: 👍 We can expose this.

File: services/app-api/handlers/reports/fetch.ts (hunk 3 above)
Comment (braxex) [reply services/app-api/handlers/reports/fetch.ts:22]: 👍 Good call.

File: services/ui-src/src/components/cards/EntityCard/EntityCard.test.tsx (hunk 2 above)
Comment (tblackwe) [reply services/ui-src/src/components/cards/EntityCard/EntityCard.test.tsx:181]: fixed

File: README.md (hunk 1 above)
Comment (tblackwe) [reply README.md]: left alone per discussion

File: services/app-api/handlers/reports/fetch.ts (hunk 3 above)
Comment (tblackwe) [reply services/app-api/handlers/reports/fetch.ts]: completed

**Pull request merged**